monkeytypes/base_models.py

```diff
@@ -1,61 +1,46 @@
-import json
-from collections.abc import Sequence
+import re
+from typing import Any
+from pydantic import BaseModel, ValidationError, ConfigDict
+from .errors import IllegalMutationError
 
-from pydantic import BaseModel, Extra, ValidationError
+TYPE_ERROR_LIST = [r"\w+_type", "int_from_float", "is_instance_of", "is_subcass_of"]
+ILLEGAL_MUTATION_LIST = ["frozen_field", "frozen_instance"]
 
 
-class InfectionMonkeyModelConfig:
-    allow_mutation = False
-    underscore_attrs_are_private = True
-    extra = Extra.forbid
+InfectionMonkeyModelConfig = ConfigDict(frozen=True, extra="forbid")
 
-
-class MutableInfectionMonkeyModelConfig(InfectionMonkeyModelConfig):
-    allow_mutation = True
-    validate_assignment = True
+MutableInfectionMonkeyModelConfig = ConfigDict(frozen=False, validate_assignment=True)
```

Comment:
This is missing `extra="forbid"`. This needs to extend what's in `InfectionMonkeyModelConfig` so that future changes tO `InfectionMonkeyModelConfig` propagate to `MutableInfectionMonkeyModelConfig`.